src/main/java/org/ehrbase/fhirbridge/ehr/converter/sofascore/SofaScoreObservationConverter.java

```diff
@@ -119,9 +175,16 @@ private void mapHerzKreislaufSystemCode(SOFAScoreObservation sofaScore, String h
 
     }
 
+    // respiration
     private void mapAtemtaetigkeitCode(SOFAScoreObservation sofaScore, Observation observation) {
-        String atemtaetigkeitCode = observation.getComponent().get(0).getValueCodeableConcept().
-                getCoding().get(0).getCode();
+
+        Observation.ObservationComponentComponent resp = getComponent(observation, "resp");
+
+        if (resp.getValueCodeableConcept().getCoding().isEmpty()) {
```

Comment:
All of this resp, liv(138) etc if cases can be generalised into one single method in order to avoid duplicates